src/components/raycaster.js

```diff
@@ -1,112 +1,138 @@
+    var prevIntersectedEls = this.intersectedEls.slice();
+    var intersectedEls = this.intersectedEls = [];  // Reset intersectedEls.
+    var intersections;
+    var prevCheckTime = this.prevCheckTime;
+
+    // Only check for intersection if interval time has passed.
+    if (prevCheckTime && (time - prevCheckTime < data.interval)) { return; }
+
+    // Raycast.
+    this.updateOriginDirection();
+    intersections = this.raycaster.intersectObjects(this.objects, data.recursive);
+
+    // Update intersectedEls object first in case event handlers try to inspect it.
+    intersections.forEach(function emitEvents (intersection) {
+      intersectedEls.push(intersection.object.el);
+    });
 
-      // If the intersected object is the cursor itself
-      // or the object is further than the max distance
+    // Emit intersected on intersected entity per intersected entity.
+    intersections.forEach(function emitEvents (intersection) {
+      var intersectedEl = intersection.object.el;
+      intersectedEl.emit('raycaster-intersected', {el: el, intersection: intersection});
+    });
 
-      if (intersectedObj.object.el === undefined) { continue; }
-      if (intersectedObj.object.el === cursorEl) { continue; }
-      if (!intersectedObj.object.visible) { continue; }
-      return intersectedObj;
+    // Emit all intersections at once on raycasting entity.
+    if (intersections.length) {
+      el.emit('raycaster-intersection', {
+        els: intersections.map(function getEl (intersection) {
+          return intersection.object.el;
+        }),
+        intersections: intersections
+      });
     }
-    return null;
-  },
 
-  /**
-   * Remembers the last intersected element
-   */
-  setExistingIntersection: function (el, distance) {
-    this.intersectedEl = el;
-    this.el.emit('intersection', { el: el, distance: distance });
+    // Emit intersection cleared on both entities per formerly intersected entity.
+    prevIntersectedEls.forEach(function checkStillIntersected (intersectedEl) {
+      if (intersectedEls.indexOf(intersectedEl) !== -1) { return; }
+      el.emit('raycaster-intersection-cleared', {el: intersectedEl});
+      intersectedEl.emit('raycaster-intersected-cleared', {el: el});
+    });
   },
 
   /**
-   * Emits a `mouseleave` event and clears info about the last intersection.
+   * Set origin and direction of raycaster using entity position and rotation.
    */
-  clearExistingIntersection: function () {
-    var intersectedEl = this.intersectedEl;
-    this.el.emit('intersectioncleared', { el: intersectedEl });
-    this.intersectedEl = null;
-  },
+  updateOriginDirection: function () {
+    var el = this.el;
+    var object3D = el.object3D;
 
-  handleIntersection: function (obj) {
-    var el = obj.object.el;
+    // Update matrix world.
+    object3D.updateMatrixWorld();
+    // Grab the position and rotation.
+    object3D.matrixWorld.decompose(this.originVec3, this.directionHelper, scaleDummy);
```

Comment:
It doesn't matter that directionHelper is used by multiple raycaster because you only use the variable during the duration of the function call. There won't be multiple instances of the same function running at the same time and they cannot interrupt each other.